internal/config_extractor.go

```diff
@@ -0,0 +1,71 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func ExtractConfigDump(checkpointPath string) (namespace, pod, container string, timestamp time.Time, err error) {
+	tempDir, err := os.MkdirTemp("", "extracted-checkpoint")
+	if err != nil {
+		return "", "", "", time.Time{}, err
+	}
+	defer os.RemoveAll(tempDir)
+
+	filesToExtract := []string{"config.dump"}
+	if err := UntarFiles(checkpointPath, tempDir, filesToExtract); err != nil {
+		log.Printf("Error extracting files from archive %s: %v\n", checkpointPath, err)
+		return "", "", "", time.Time{}, err
+	}
+
+	configDumpPath := filepath.Join(tempDir, "config.dump")
+	content, err := os.ReadFile(configDumpPath)
+	if err != nil {
+		log.Printf("Error reading config.dump file: %v\n", err)
+		return "", "", "", time.Time{}, err
+	}
+
+	return extractConfigDumpContent(content)
+}
+
+type Content struct {
```

Comment:
You should use https://github.com/checkpoint-restore/checkpointctl/blob/main/lib/metadata.go#L38

Reply:
yes I update that.